ENIGMAsystem/SHELL/Platforms/General/PFwindow.cpp

```diff
@@ -287,8 +287,9 @@ void window_default(bool center_size) {
 }
 
 void window_center() {
-  window_set_position(display_get_width() / 2 - window_get_width() / 2,
-                      display_get_height() / 2 - window_get_height() / 2);
+  // without this while loop, the window won't be guaranteed to center at game init, in xlib, which is weird, but this makes it necessary.
+  while (enigma::windowX != display_get_width() / 2 - window_get_width() / 2 || enigma::windowY != display_get_height() / 2 - window_get_height() / 2)
```

Comment:
Why is this a while loop?

Reply:
What do you mean 'bound this'?

Reply:
On a similar topic, you never answered my question on discord if this for loop might loop forever on titling window managers and other similar circumstances, I might as well address that in this pr as well if you're game: https://github.com/enigma-dev/enigma-dev/blob/c9352a7e1a52b0add48543ed76c01ad12bca9c83/ENIGMAsystem/SHELL/Platforms/xlib/XLIBwindow.cpp#L325

Reply:
Alright, I took care of it. I'll let you know when it passes. Thanks!